### ISSUE TITLE
Scheduler tasks stuck

### ISSUE THREAD

**Issue by timur560** [webtoolsnz/yii2-scheduler#3]
Once there was a problem in running task (some fatal error maybe, or server issues), field 'started_at' not clear and task will stuck and never run again.

**Comment by marius-nabal** [comment]
I think that was there intentionally so that a "running" task wouldn't run again.  If a task does not end normally and is left in the running state, "usually" it's better to not simply try to rerun the task, but to investigate the cause of the problem.
There used to be a facility to manually force a task to re-run - which could be done after the problem has been rectified.
Maybe take the above into consideration when you refactor.

**Comment by timur560** [comment]
To not re-run task it is great idea, sure, but if my task stops for a few days while I didn't see that it is not running, it is a real problem. It's useful to implement some mechanism to avoid that. Or at least show in web interface that task is stacked too long time and some button to re-run it manually.
Thank you.

**Comment by marius-nabal** [comment]
Yes, if you can think of a good way to do that, it would be great.  Maybe run the idea by Byron. Scheduler is a very useful component and any improvements will be welcomed!

**Comment by badams** [comment]
@timur560 I agree the current way this is handled is not correct, a shutdown handler should update the state of the task before exiting, this way we can see the task is failing.

I think its fine to let it keep failing, as the failure could be caused by an external issue, such as a 3rd party service outage; which could remedy itself.

**Comment by badams** [comment]
@timur560 this is has been addressed in release [0.0.12](https://github.com/webtoolsnz/yii2-scheduler/releases/tag/0.0.12)

You will need to update your console config file with the following:

```
    'components' => [
        'errorHandler' => [
            'class' => 'webtoolsnz\scheduler\ErrorHandler'
        ]
    ]
```

**Comment by timur560** [comment]
Very useful fast update. Thank you, @badams .
